src/sql/src/plan/statement/ddl.rs

```diff
@@ -1223,10 +1230,30 @@ pub fn plan_create_source(
             });
             (key_desc, value_desc)
         }
-        None => (None, external_connection.value_desc()),
+        None => (
+            Some(external_connection.key_desc()),
+            external_connection.value_desc(),
+        ),
     };
 
-    let mut key_envelope = get_key_envelope(include_metadata, encoding.as_ref())?;
+    // KEY VALUE load generators are the only UPSERT source that
+    // has no encoding but defaults to `INCLUDE KEY`.
+    //
+    // TODO(guswynn|petrosagg): when this is refactored, remove this special-case
+    // and move the `bail_unsupported` call below to a check specific to planning
+    // the decode stage.
+    let key_envelope_no_encoding = matches!(
```

Comment:
I don't think we need this. If you use the key value load generator without `ENVELOPE UPSERT` then you should explicitly specify `INCLUDE KEY` if you want to see the key. When you do specify `ENVELOPE UPSERT` then the planner already auto-sets `INCLUDE KEY` if not specified by the user

Reply:
The code that does that panics; it relies on an `encoding` being available :(

Reply:
I looked at this closer today:

Note that the code as it stands does the behavior you specify: `INCLUDE KEY` is not auto-set, unless its specified by the user OR they use `ENVELOPE UPSERT`. This is tested at the bottom on the td file in the pr.

Removing this special case boils down to removing these errors:
```
> INCLUDE KEY requires specifying KEY FORMAT .. VALUE FORMAT, got bare FORMAT`
> UPSERT requires a key/value format: {:?}
> ENVELOPE [DEBEZIUM] UPSERT requires that KEY FORMAT be specified
```

(These are actually all very very similar errors. There is a world where we invert the special case, and much earlier in planning assert that Kafka sources have the above requirements, but thats still leaving a special case in), which amounts to deciding to:

  default the key envelope of kafka sources to BYTES
  
Is that something we want to commit to?